### PR TITLE
Announce 4.65

### PR DESCRIPTION
**Pull request by fredemmott**

### CODE REVIEW

File: _posts/2020-07-06-hhvm-4.65.markdown [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  `preg_replace_with_error()`, to avoid the need for `preg_last_error()` calls
+  and the global state that entails.
+- added `HH\is_vec_or_varray()` and `HH\is_dict_or_darray()` functions.
+- clearer error messages when incompatible kinds `extend` each other - for
+  example, a class can not `extend` an interface, but it can `implement` one.
+
+# Breaking Changes
+
+- `array()` literals are no longer supported by the parser; `[]` literals were
+  already banned. Replace with `darray[]` or `varray[]`.
```
Comment (jjergus) [inline _posts/2020-07-06-hhvm-4.65.markdown:28]: Is it worth mentioning the HHAST migration again? (it was mentioned in future changes last week)

File: _posts/2020-07-06-hhvm-4.65.markdown [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- the globals array has been removed - for example, `HH\global_get('GLOBALS')`
+  now returns null. Fetch specific known keys instead.
```
Comment (jjergus) [inline _posts/2020-07-06-hhvm-4.65.markdown:30]: Probably worth mentioning that `$GLOBALS['some_key']` still works (or doesn't it)?

Comment (fredemmott) [reply _posts/2020-07-06-hhvm-4.65.markdown:30]: That seems like an oversight to me - will follow up:

```
hphpd> =$GLOBALS['foo']
=$GLOBALS['foo']
null
hphpd> =$GLOBALS
=$GLOBALS
Hit a php exception : Error: Access $GLOBALS via wrappers in :1
hphpd> =$waefawef['foo']
=$waefawef['foo']

Notice: Undefined variable: waefawef
null
```

File: _posts/2020-07-06-hhvm-4.65.markdown [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  and the global state that entails.
+- added `HH\is_vec_or_varray()` and `HH\is_dict_or_darray()` functions.
+- clearer error messages when incompatible kinds `extend` each other - for
+  example, a class can not `extend` an interface, but it can `implement` one.
+
+# Breaking Changes
+
+- `array()` literals are no longer supported by the parser; `[]` literals were
+  already banned. Replace with `darray[]` or `varray[]`. On older versions of
+  HHVM, `hhast-migrate --php-arrays` can be used to automate this.
```
Comment (Wilfred) [inline _posts/2020-07-06-hhvm-4.65.markdown:29]: This is news to me, albeit very good news :)

Have we changed the default parsing options, or is the parsing logic entirely removed? Could you point me at the diff?

Comment (jjergus) [reply _posts/2020-07-06-hhvm-4.65.markdown:29]: I think the code is still there but turned off by default everywhere.

- runtime: https://github.com/facebook/hhvm/commit/2f549ff4a4b3d0fa4c27a438351edc556dd7efe1
- parser: https://github.com/facebook/hhvm/commit/38a1920c6826f065584ee5fbf121fe22811a3624

Comment (fredemmott) [reply _posts/2020-07-06-hhvm-4.65.markdown:29]: I wasn't able to figure out a way to do this through hhconfig; I didn't investigate spend much time on it as AIUI Daniel is planning to actually delete the code this week

**Pull request merged**